pennylane/qinfo/fisher.py

```diff
@@ -236,3 +237,27 @@ def _make_probs(tape, wires=None, post_processing_fn=None):
         post_processing_fn = lambda x: qml.math.squeeze(qml.math.stack(x))
 
     return [new_tape], post_processing_fn
+
+
+def quantum_fisher(*args, **kwargs):
+    r"""Returns a function that computes the quantum fisher information matrix (QFIM) of a given :class:`.QNode` or quantum tape.
+
+    Given a parametrized quantum state :math:`|\psi(\bm{\theta})\rangle`, the quantum fisher information matrix (QFIM) quantifies how changes to the parameters :math:`\bm{\theta}`
+    are reflected in the quantum state. The metric used to induce the QFIM is the fidelity :math:`f = |\langle \psi | \psi' \rangle|^2` between two (pure) quantum states.
+    This leads to the following definition of the QFIM (see eq. (27) in `arxiv:2103.15191 <https://arxiv.org/abs/2103.15191>`_):
+
+    .. math::
+
+        \text{QFIM}_{i, j} = 4 \text{Re}\left[ \langle \partial_i \psi(\bm{\theta}) | \partial_j \psi(\bm{\theta}) \rangle
+        - \langle \partial_i \psi(\bm{\theta}) | \psi(\bm{\theta}) \rangle \langle \psi(\bm{\theta}) | \partial_j \psi(\bm{\theta}) \rangle \right]
+
+    with short notation :math:`| \partial_j \psi(\bm{\theta}) \rangle := \frac{\partial}{\partial \theta_j}| \psi(\bm{\theta}) \rangle`.
+
+    .. seealso::
+        :func:`~.pennylane.adjoint_metric_tensor`; ``quantum_fisher()`` is simply calling :func:`~.pennylane.metric_tensor` with a prefactor of 4. Please refer there for implementation details.
+    """
+
```

Comment:
Would be good to add a check here for the shots in tape, such that `metric_tensor` can be passed if they are finite and `adjoint_metric_tensor` when they are None. Is that possible for transforms like these?

Reply:
That's a really good idea! This isn't currently possible, but we would like to overhaul how shots are architected internally, so that this does become possible.

Currently, shots are stored at the device, not at the transform level

Reply:
So for the moment maybe add a boolean variable `hardware` (or something better) and just to an if-clause?